src/NHibernate/Mapping/DenormalizedTable.cs

```diff
@@ -43,6 +43,12 @@ public override IEnumerable<Index> IndexIterator
 				IEnumerable<Index> includedIdxs = includedTable.IndexIterator;
 				foreach (Index parentIndex in includedIdxs)
 				{
+					var sharedIndex = GetIndex(parentIndex.Name);
+					if (sharedIndex != null)
+					{
+						sharedIndex.AddColumns(parentIndex.ColumnIterator);
+						continue;
+					}
 					Index index = new Index();
 					index.Name = Name + parentIndex.Name;
```

Comment:
With this fix it seems  this change is not required and we can keep existing index name intact. Should we?

Reply:
Ok. I see why this change is needed. Some of DBs like PostgreSQL need unique index name...